core/variant/dictionary.h

```diff
@@ -46,7 +46,7 @@ class Dictionary {
 	void _unref() const;
 
 public:
-	void get_key_list(List<Variant> *p_keys) const;
+	void get_key_list(List<Variant> *p_keys, bool p_string_name_to_string = false) const;
```

Comment:
New alternative solution which fetches keys as `String` when `StringName` was found, replacing the custom sort (doesn't solve the visual shader case though)